functions/Set-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,302 @@
+                            Stop-Function -Message ("The database is not present on instance $instance.") -Target $SqlServer 
+                            return
+                        }
+                    }
+
+                    if (($DatabaseUserName) -and ($DatabaseName)) {
+                        # Check if the username is present in the database
+                        if (($Server.Databases[$DatabaseName].Users).Name -contains $DatabaseUserName) {
+                            Write-Message -Message "Setting job step database username to $DatabaseUserName" -Level Verbose 
+                            $JobStep.DatabaseUserName = $DatabaseUserName
+                        }
+                        else {
+                            Stop-Function -Message ("The database user is not present in the database $DatabaseName on instance $instance.") -Target $SqlServer 
+                            return
+                        }
+                    }
+
+                    if ($RetryAttempts) {
+                        Write-Message -Message "Setting job step retry attempts to $RetryAttempts" -Level Verbose 
+                        $JobStep.RetryAttempts = $RetryAttempts
+                    }
+
+                    if ($RetryInterval) {
+                        Write-Message -Message "Setting job step retry interval to $RetryInterval" -Level Verbose 
+                        $JobStep.RetryInterval = $RetryInterval
+                    }
+
+                    if ($OutputFileName) {
+                        Write-Message -Message "Setting job step output file name to $OutputFileName" -Level Verbose 
+                        $JobStep.OutputFileName = $OutputFileName
+                    }
+
+                    if ($ProxyName) {
+                        # Check if the proxy exists
+                        if (($Server.JobServer.ProxyAccounts).Name -contains $ProxyName) {
+                            Write-Message -Message "Setting job step proxy name to $ProxyName" -Level Verbose 
+                            $JobStep.ProxyName = $ProxyName
+                        }
+                        else {
+                            Stop-Function -Message ("The proxy name $ProxyName doesn't exist on instance $instance.") -Target $SqlServer 
+                            return
+                        }
+                    }
+                    #region job step options
+
+                    # Execute 
+                    if ($PSCmdlet.ShouldProcess($SqlServer, ("Changing the job step $StepName"))) {
+                        try {
+                            Write-Message -Message ("Changing the job step") -Level Output 
+                        
+                            # Change the job step
+                            $JobStep.Alter()
+                        }
+                        catch {
+                            Write-Message -Message ("Something went wrong changing the job step. `n$_.Exception.Message)") -Level Output 
```

Comment:
Should use `Stop-Function` with the ` -Continue`  parameter instead. Avoid artificial linebreaks and remember that the properties of variables are not evaluated in strings. So rather than this:
`"$_.Exception.Message"`
use this:
`"$($_.Exception.Message)"`